sklearn/impute/_iterative.py

```diff
@@ -383,7 +391,10 @@ def _get_ordered_idx(self, mask_missing_values):
             The order in which to impute the features.
         """
         frac_of_missing_values = mask_missing_values.mean(axis=0)
-        missing_values_idx = np.nonzero(frac_of_missing_values)[0]
+        if self.skip_complete:
+            missing_values_idx = np.nonzero(frac_of_missing_values)[0]
```

Comment:
```suggestion
            missing_values_idx = np.flatnonzero(frac_of_missing_values)
```

Reply:
Didn't know about this function, thanks!